### ISSUE TITLE
PDF file coming back as completely white/transparent png

### ISSUE THREAD

**Issue by dontwork** [imgproxy/imgproxy#554]
Hi, this issue is in reference to 2 pdf files supplied previously.

The pdfs are being requested from imgproxy as pngs with the the config of `/fit/1050/1485/no/0`

The images returned from imgproxy are completely blank. Other pdfs we have accessed via imgproxy with the same configuration have been successful

**Comment by DarthSim** [comment]
That was rough, but I managed to debug and fix this. Check out the `latest` build.

**Comment by dontwork** [comment]
@DarthSim Hi, I have deployed with FROM eu.gcr.io/pw-artwork-upload/imgproxy:latest and the issue remains. Let me know if I can provide you with anything

**Comment by dontwork** [comment]
You have nailed it! Working 100% as expected. Would love to understand what it was about this specific PDF that was causing issues if you have the time to share.

Thanks very much!

**Comment by DarthSim** [comment]
Well, I don't completely understand that either. The reason for the issue was that libvips sets a white background for PDFs by default by filling the initial buffer with white pixels. Usually, this works as intended but in this case, the PDF is rendered pretty weirdly using the background in a very strange way. With a white background, the result becomes completely white, with a black one, the result becomes completely black, but with a gray one, I can see the source image grayed out. It looks like the shapes in the PDF are drawn in some fancy way using masking or something like that.

There was another weird thing that killed some of my time. I didn't notice from the start that libvips fills the initial buffer with white. So I tried to fill it using PDFium API, but somehow this didn't help at all. When I had filled the canvas with transparent pixels, the result remained completely white. When I made libvips to initialize the buffer with transparent pixels, everything started to work fine.

Another thing that took some time was that I started digging from the wrong end. The background thing was my least suspect, and I tried to rebuild PDFium enabling some of its features, and dissected the PDFium example app just to find out that there's nothing wrong with PDF rendering code itself.
